code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_core.py

```diff
@@ -0,0 +1,290 @@
+            profile (CyclerDataProfileC): [description]
+
+        Returns:
+            bool: [description]
+        """
+        # Check if the profile range is inside the battery range
+        res = False
+        if profile.range.no_voltage() and profile.range.no_current():
+            log.debug("Profile no current neither voltages")
+            res = True
+        elif profile.range.no_current():
+            log.debug("Profile no current")
+            res = profile.range.in_range_voltage(battery.elec_ranges)
+        elif profile.range.no_voltage():
+            log.debug("Profile no voltage")
+            res = profile.range.in_range_current(battery.elec_ranges)
+        else:
+            log.debug("Profile with current and voltage")
+            res = (profile.range.in_range_current(battery.elec_ranges) and
+                profile.range.in_range_voltage(battery.elec_ranges))
+        log.critical(f"Validating experiment: {res}")
+        return res
+
+    def __update_exp_status(self, exp_status: CyclerDataExpStatusE) -> None:
+        """Write the experiment status in the shared memory
+
+        Args:
+            exp_status (CyclerDataExpStatusE): Experiment status
+        """
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.SET_EXP_STATUS,
+                        exp_status= exp_status)
+        self.__chan_str_reqs.send_data(request)
+
+    def turn_deprecated(self):
+        """Send a deprecated command to the str node to turn all experiments to error
+        """
+        request: MidStrCmdDataC = MidStrCmdDataC(cmd_type= MidStrReqCmdE.TURN_DEPRECATED)
+        self.__chan_str_reqs.send_data(request)
+
+    def __execute_experiment(self) -> None:
+        """Execute the experiment
+        """
+        # First step is to update the local data in power
+        self.exp_status, self.__local_gen_meas.instr_id = self.pwr_control.process_iteration()
+        self.__update_exp_status(self.exp_status)
+
+    def execute_machine_status(self) -> None: #pylint: disable=too-many-branches
+        """Execute the machine status
+        """
+        log.debug("Executing machine status")
+        try:
+            ## Process machine status
+            if self.state == AppManCoreStatusE.GET_EXP:
+                ## Wait for cs status to continue
+                if ((not self.__wait_cs_reqst and not self.__wait_exp_reqst) or
+                    self.__iter> _PERIOD_WAIT_EXP):
+                    self.__fetch_new_exp()
+                    self.__request_cs_status()
+                ## Check if the cycler station is deprecated
+                if self.is_deprecated:
+                    log.critical("Cycler station is deprecated")
+                    self.turn_deprecated()
+                    self.state = AppManCoreStatusE.ERROR
+                ## Wait for the experiment and check if the experiment is valid
+                if not self.experiment is None and self.__wait_exp_reqst:
```

Comment:
Change to self.experiment is not None and ....

Reply:
and not wait_exp_reqst

Reply:
Done